ui/packages/consul-ui/app/routes/application.js

```diff
@@ -13,13 +13,21 @@ export default class ApplicationRoute extends Route.extend(WithBlockingActions)
   @service('client/http') client;
   @service('env') env;
   @service() hcp;
+  @service() router;
 
   data;
 
+  beforeModel() {
+    if (this.env.var('CONSUL_V2_CATALOG_ENABLED')) {
+      this.router.replaceWith('unavailable');
+    }
+  }
+
   async model() {
-    if (this.env.var('CONSUL_ACLS_ENABLED')) {
+    if (this.env.var('CONSUL_ACLS_ENABLED') && !this.env.var('CONSUL_V2_CATALOG_ENABLED')) {
```

Comment:
Do we need this check - `CONSUL_V2_CATALOG_ENABLED` always be false here?
Won't the `beforeModel` take us to `unavailable`?

Reply:
Ya you are correct. Good suggestion, I'll update it on Monday :)